### PR TITLE
Move Exercises to React

### PR DESCRIPTION
**Pull request by iHiD**

### CODE REVIEW

**Comment by kntsoriano** [comment]
@iHiD I just pushed the commits to this branch. We should be able to have the exercise widget and exercise list component working.

Currently, the exercise widget component is in a weird state where it tries to do two things:
1. Preserve it's old functionality (accepting `size`, `desc`)
2. Implement new functionality (passing in the solutions, etc)

I think you'd be able to decide on how to move forward with it. I just put it all in there right now.

**Comment by iHiD** [comment]
@kntsoriano Should I work on this from here?

**Comment by iHiD** [comment]
@kntsoriano. I've updated this with the right styling. I changed quite a bit of the React in the widget to get it correct - feel free to rearrange things if you think I've done anything wrong.

Three jobs remaining:
1. Add `--recommended` className to `c-exercise-widget --available` if `exercise.recommended`
2. Please check the code
3. Please get some good test coverage for both the exercise-widget and the exercise list.

Then let's get this merged! :)

**Comment by iHiD** [comment]
@kntsoriano I've now updated this further. Can you implement the tooltip in JS for Exercise too pls. It should work the same as the Concept one (just render the returned HTML). The endpoint is there (tooltip_track_exercise_path) and I've added it to one sample place for you to test from.

File: app/views/tracks/show/joined.html.haml [hunk 1]
```diff
@@ -30,7 +30,7 @@
           .exercises
             - @exercise_statuses.each do |exercise_slug, status|
               - if status == :locked
-                .c-ed.--l
+                .c-ed.--l{ "data-tooltip-type": "exercise", "data-endpoint": tooltip_track_exercise_path(@track, exercise_slug) }
```
Comment (iHiD) [inline app/views/tracks/show/joined.html.haml]: @kntsoriano It should work here.

**Comment by kntsoriano** [comment]
@iHiD Should be ready :) Can you fix the merge conflicts for  app/views/tracks/show/joined.html.haml?

**Comment by iHiD** [comment]
Even the system tests went green!!

**Pull request merged**